### PR TITLE
primer3-2.4.0: migrate

### PR DESCRIPTION
**Pull request by marc-reh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/primer3.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  bottle do
+    cellar :any_skip_relocation
+    sha256 "bcc7d166ca32302dca6650cd4097255e7dd2dd3f890a1a864498586b71a44a3e" => :high_sierra
+    sha256 "fe08ddd1929b113d59912a7b86bc90eb0dcd47873b2d00263609b685b45c4973" => :sierra
+    sha256 "e5c5bd6002dd36aa1fa177dc73bcb2feeae9b7a421a6ecbac4443866827378ae" => :el_capitan
+    sha256 "62595ddbfc162732a7d4741b1c331ef51d53fe8ac2008c613447bb4ebb7a2b1c" => :x86_64_linux
+  end
+
+  option "without-test", "Skip build-time tests"
```
Comment (fxcoudert) [inline Formula/primer3.rb]: We don't run tests except for security software. Let's more this option.

File: Formula/primer3.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    cd "src" do
+      system "make"
+      system "make", "test" if build.with? "test"
+      bin.install %w[primer3_core ntdpal ntthal oligotm long_seq_tm_test]
+      pkgshare.install "primer3_config"
+    end
+  end
+
+  test do
+    system "#{bin}/long_seq_tm_test", "AAAAGGGCCCCCCCCTTTTTTTTTTT", "3", "20"
```
Comment (fxcoudert) [inline Formula/primer3.rb]: Can we check the output?

File: Formula/primer3.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 "62595ddbfc162732a7d4741b1c331ef51d53fe8ac2008c613447bb4ebb7a2b1c" => :x86_64_linux
+  end
+
+  option "without-test", "Skip build-time tests"
+
+  def install
+    cd "src" do
+      system "make"
+      system "make", "test" if build.with? "test"
+      bin.install %w[primer3_core ntdpal ntthal oligotm long_seq_tm_test]
```
Comment (fxcoudert) [inline Formula/primer3.rb]: Lack of `make install` target reported to upstream: https://github.com/primer3-org/primer3/issues/1
Please add a comment and link above that line

File: Formula/primer3.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 "e5c5bd6002dd36aa1fa177dc73bcb2feeae9b7a421a6ecbac4443866827378ae" => :el_capitan
+    sha256 "62595ddbfc162732a7d4741b1c331ef51d53fe8ac2008c613447bb4ebb7a2b1c" => :x86_64_linux
+  end
+
+  option "without-test", "Skip build-time tests"
+
+  def install
+    cd "src" do
+      system "make"
+      system "make", "test" if build.with? "test"
```
Comment (fxcoudert) [inline Formula/primer3.rb]: Remove `make test`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/primer3.rb (hunk 1 above)
Comment (marc-reh) [reply Formula/primer3.rb]: done

File: Formula/primer3.rb (hunk 3 above)
Comment (marc-reh) [reply Formula/primer3.rb]: done

File: Formula/primer3.rb (hunk 4 above)
Comment (marc-reh) [reply Formula/primer3.rb]: done

File: Formula/primer3.rb (hunk 2 above)
Comment (marc-reh) [reply Formula/primer3.rb]: done

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @marc-reh for the pull request!

**Pull request closed without merging**